.github/workflows/formatting.yml

```diff
@@ -26,3 +26,24 @@ jobs:
       - name: Check formatting
         working-directory: ${{runner.workspace}}/nrn
         run: external/coding-conventions/bin/format -v --dry-run
+
+      # If formatting fails, apply formatting and push changes.
```

Comment:
@alexsavulescu : the topic of auto-formatting was discussed in the past: As there is possibility of false positives / auto-formatting is susceptible to subtle issues, the decision was not to apply auto formatting.

Reply:
I see. Don't remember any details around that discussion however. Concerning those subtle issues, they would still need to be addressed, regardless of auto-formatting? Just considering the number of `make format happy` commits out there ..

Reply:
at least what I remember is, doing "clang-format" automatically on the code base was not the preferred option. Maybe @olupton @matz-e or @ohm314 might have an opinion.

Reply:
Ok. having some issues with this PR so putting it aside until we clear it.

Reply:
remind me again: if I am not mistaken, here we are applying auto-formatting and pushing the changes *again to the same PR branch*, right?